app/models/service_provider_request.rb

```diff
@@ -1,16 +1,27 @@
 class ServiceProviderRequest
   attr_accessor :uuid, :issuer, :url, :ial, :aal, :requested_attributes
 
-  def initialize(uuid: nil, issuer: nil, url: nil, ial: nil, aal: nil, requested_attributes: [])
+  def initialize(uuid: nil, issuer: nil, url: nil, loa: nil, ial: nil, aal: nil, requested_attributes: [])
     @uuid = uuid
     @issuer = issuer
     @url = url
-    @ial = ial
+    @ial = ial || loa
     @aal = aal
     @requested_attributes = requested_attributes&.map(&:to_s)
+    Rails.logger.info { 'Note: loa used to initialize ServiceProviderRequest' } if loa
```

Comment:
block precedence is super weird in Ruby, I'm worrried about that trailing `if` with it.....can we use the direct argument syntax?

```
Rails.logger.info("Note: loa used to....") if loa
```